metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala

```diff
@@ -0,0 +1,161 @@
+package scala.meta.internal.troubleshoot
+
+import scala.collection.mutable.ListBuffer
+
+import scala.meta.internal.metals.Messages
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ScalaTarget
+import scala.meta.internal.metals.ScalaVersions
+import scala.meta.io.AbsolutePath
+
+class ProblemResolver(
+    workspace: AbsolutePath,
+    isUnsupportedBloopVersion: () => Boolean,
+    bloopVersion: () => Option[String],
```

Comment:
Mentioned this in `Doctor.scala`, but I think it might be cleaner to only have `bloopVersion` here and then move the `isUnsupportedBloopVersion` method from `Doctor.scala` into here and have the logic to determine whether or not it's unsupported in here since a lot of the other logic for finding various problems is in here.